pkg/archive/filter.go

```diff
@@ -46,7 +46,14 @@ func tryProcFilter(args []string, input io.Reader, cleanup func()) (io.ReadClose
 	go func() {
 		err := cmd.Run()
 		if err != nil && stderrBuf.Len() > 0 {
-			err = fmt.Errorf("%s: %w", strings.TrimRight(stderrBuf.String(), "\n"), err)
+			b := make([]byte, 0)
```

Comment:
The `0`-byte slice is extremely unusual — worth a detailed comment I think.

Reply:
Or, maybe, read from a 1-byte slice; that has a higher chance of reporting an error … and _maybe_ a higher chance of hanging? But it seems to me that any `Read` that is going to special-case zero-byte reads is also going to report fewer errors on that special-case path.

The 1-byte slice seems attractive to me in that it would be a much more ordinary code.

Reply:
changed to 1 byte